OpenRA.Mods.Common/Activities/Move/MoveOnto.cs

```diff
@@ -0,0 +1,55 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class MoveOnto : MoveAdjacentTo
+	{
+		readonly WVec offset = WVec.Zero;
+
+		public MoveOnto(Actor self, in Target target, WVec? offset = null, WPos? initialTargetPosition = null, Color? targetLineColor = null)
+			: base(self, target, initialTargetPosition, targetLineColor)
+		{
+			if (offset.HasValue)
+				this.offset = offset.Value;
+		}
+
+		protected override void SetVisibleTargetLocation(Actor self, Target target)
+		{
+			lastVisibleTargetLocation = self.World.Map.CellContaining(Target.CenterPosition + offset);
+		}
+
+		protected override bool ShouldStop(Actor self)
+		{
+			// If we are right next to the Actor and target type is still not Actor, we assume the target is dead.
+			return Target.Type != TargetType.Actor && Util.AreAdjacentCells(lastVisibleTargetLocation, self.Location);
```

Comment:
This looks a bit weird too. Why should we not stop earlier if the target (host) is dead and we observed that? Shouldn't this automatically happen inside the base `Tick` method anyway? (It already handles targets that are not visible.)

Reply:
If we observed that then the target type won't be Actor

Reply:
> Why should we not stop earlier if the target (host) is dead and we observed that? 

That's the current behaviour.  I'm not sure wether we want to change it

Reply:
> If we observed that then the target type won't be Actor

Yes but what is the point of this check then? If we observed it we should not stop only when next to it, right? We probably even already stop in the Tick method.

Reply:
> This check is for units to not cram into a single cell if there's nothing to cram into

Reply:
And we're back to my original question: Units should/would have stopped already, so why is this necessary?

Reply:
Changed to `Target.Type == TargetType.Terrain` as requested on discord